custom_components/better_thermostat/climate.py

```diff
@@ -910,18 +919,12 @@ async def _async_control_heating(self):
 				self._active = True
 				self.ignore_states = True
 				# Use the same precision and min and max as the TRV
-				if self.hass.states.get(self.heater_entity_id).attributes.get('target_temp_step') is not None:
+				if self.hass.states.get(self.heater_entity_id).attributes.get('target_temp_step') is not None and self.target_temperature_step is None:
 					self._TRV_target_temp_step = float(self.hass.states.get(self.heater_entity_id).attributes.get('target_temp_step'))
-				else:
-					self._TRV_target_temp_step = 1
```

Comment:
Why did you delete this code? If the heater reports no target_temp_step we should assume integer steps and not ignore that - shouldn't we?